### PR TITLE
fix(camera): update permission logic for get_snapshot method

### PR DESCRIPTION
**Pull request by RaHehl**
Allow get_snapshot to work when either READ_LIVE or READ_MEDIA permissions are granted, depending on the context.

### Description of change

Fixes https://github.com/home-assistant/core/issues/131900

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] This pull request follows the [contributing guidelines](https://github.com/uilibs/uiprotect/blob/main/CONTRIBUTING.md).
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/), such as "fix(api): prevent racing of requests".

> - If pre-commit.ci is failing, try `pre-commit run -a` for further information.
> - If CI / test is failing, try `poetry run pytest` for further information.

## Summary by CodeRabbit

- **New Features**
	- Improved permission logic for accessing camera snapshots, allowing more flexible authorization.
	- Enhanced error messages for unauthorized access to provide clearer context.

- **Bug Fixes**
	- Updated tests to accurately reflect the new permission checks and error handling for camera snapshot retrieval.

- **Documentation**
	- Renamed test functions for clarity and added new tests to cover additional permission scenarios.

### CODE REVIEW

**Comment by RaHehl** [comment]
-> Draft: forgot the package camera

**Comment by RaHehl** [comment]
@coderabbitai review

**Comment by RaHehl** [comment]
@coderabbitai review

File: tests/data/test_camera.py [hunk 1]
```diff
@@ -1445,3 +1494,43 @@ async def test_get_package_snapshot_with_dt(camera_obj: Camera):
...
+async def test_get_package_snapshot_dt_no_read_media(camera_obj: Camera | None):
+    camera_obj._api = MagicMock(spec=ProtectApiClient)
+    camera_obj._api.get_package_camera_snapshot = AsyncMock(
+        return_value=b"snapshot_data"
+    )
+    camera_obj.feature_flags.has_package_camera = True
+
+    auth_user = camera_obj._api.bootstrap.auth_user
+
+    def mock_can(model_type, permission, camera):
```
Comment (bdraco) [inline tests/data/test_camera.py:1526]: Please type this mock

File: tests/data/test_camera.py [hunk 2]
```diff
@@ -1387,42 +1401,77 @@ async def test_get_snapshot_with_dt(camera_obj: Camera):
...
     camera_obj._api.get_package_camera_snapshot = AsyncMock(
         return_value=b"snapshot_data"
     )
+    camera_obj.feature_flags.has_package_camera = True
 
     auth_user = camera_obj._api.bootstrap.auth_user
 
-    with patch.object(auth_user, "can", return_value=False):
-        camera_obj.api.api_request.reset_mock()
+    def mock_can(model_type, permission, camera):
```
Comment (bdraco) [inline tests/data/test_camera.py:1468]: Please type this mock

File: tests/data/test_camera.py [hunk 3]
```diff
@@ -1387,42 +1401,77 @@ async def test_get_snapshot_with_dt(camera_obj: Camera):
...
+async def test_get_package_snapshot_read_media_granted(camera_obj: Camera | None):
     camera_obj._api = MagicMock(spec=ProtectApiClient)
+    camera_obj._api.get_package_camera_snapshot = AsyncMock(
+        return_value=b"snapshot_data"
+    )
     camera_obj.feature_flags.has_package_camera = True
+
+    auth_user = camera_obj._api.bootstrap.auth_user
+
+    def mock_can(model_type, permission, camera):
```
Comment (bdraco) [inline tests/data/test_camera.py:1450]: Please type this mock

File: tests/data/test_camera.py [hunk 4]
```diff
@@ -1336,36 +1336,50 @@ async def test_camera_set_ptz_home(ptz_camera: Camera | None):
...
+@pytest.mark.asyncio
+async def test_get_snapshot_no_permissions(camera_obj: Camera | None):
     camera_obj._api = MagicMock(spec=ProtectApiClient)
     camera_obj._api.get_camera_snapshot = AsyncMock(return_value=b"snapshot_data")
 
     auth_user = camera_obj._api.bootstrap.auth_user
 
-    with patch.object(auth_user, "can", return_value=False):
-        camera_obj.api.api_request.reset_mock()
+    def mock_can(model_type, permission, camera):
```
Comment (bdraco) [inline tests/data/test_camera.py:1376]: Please type this mock

File: tests/data/test_camera.py [hunk 5]
```diff
@@ -1336,36 +1336,50 @@ async def test_camera_set_ptz_home(ptz_camera: Camera | None):
...
-async def test_get_snapshot_read_live(camera_obj: Camera | None):
-    camera_obj.api.api_request.reset_mock()
+@pytest.mark.asyncio
+async def test_get_snapshot_read_live_granted(camera_obj: Camera | None):
     camera_obj._api = MagicMock(spec=ProtectApiClient)
+    camera_obj._api.get_camera_snapshot = AsyncMock(return_value=b"snapshot_data")
 
+    auth_user = camera_obj._api.bootstrap.auth_user
+
+    def mock_can(model_type, permission, camera):
```
Comment (bdraco) [inline tests/data/test_camera.py:1346]: Please type this mock

**Review by bdraco — APPROVED** [review]

**Pull request merged**